### PR TITLE
Table implementation

### PR DESCRIPTION
**Pull request by ashotvdovychenko**

### CODE REVIEW

**Review by Pagendera — APPROVED** [review]

File: scripts/form.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
+
+class SmartControl {
+    #controlRef
+
+    constructor(selector, initialValue, additionalValidators = []) {
+        this.#controlRef = document.querySelector(selector);
+
+        this.writeValue(initialValue);
+
+        additionalValidators.forEach(additionalValidator => additionalValidator(this.#controlRef))
```
Comment (Pagendera) [inline scripts/form.js]: Please, don't forget about semicolon, recheck all files and fix it

**Review by Pagendera — CHANGES_REQUESTED** [review]

File: scripts/form.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    #listenControlWriteValue() {
+        this.control.addEventListener('keydown', (event) => {
+            event.preventDefault();
+
+            const { target, key, keyCode } = event;
+            const inputValue = target.value;
+            const isNumber = keyCode > 47 && keyCode < 58;
+            const isBackspace = keyCode === 8;
```
Comment (Pagendera) [inline scripts/form.js:41]: Backspace is not working

**Review by Pagendera — CHANGES_REQUESTED** [review]

File: scripts/form.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    #getPhoneMaskWithAddedNumber(value, key) {
+        const valueArray = [...value];
+        const currentIndex = valueArray.findIndex(position => position === '_')
+
+        valueArray[currentIndex] = key;
+
+        return valueArray.join('');
+    }
+
+    #getPhoneMaskWithRemovedNumber(value, initialMask) {
```
Comment (Pagendera) [inline scripts/form.js:63]: Method is not working

**Review by Pagendera — CHANGES_REQUESTED** [review]

File: scripts/form.js (hunk 1 above)
Comment (ashotvdovychenko) [reply scripts/form.js]: Fixed

File: scripts/form.js (hunk 3 above)
Comment (ashotvdovychenko) [reply scripts/form.js:63]: Fixed

**Review by neodavis — APPROVED** [review]

**Review by Pagendera — APPROVED** [review]

**Pull request merged**